### ISSUE TITLE
Github action yml download script: "Release tag doesn't match checksum"

### ISSUE THREAD

**Issue by hjizettle** [detekt/detekt#5228]
## Expected Behavior
The detekt realease should be downloaded and and run

## Observed Behavior
The gh graphql script to download the release artifacts doesn't seem to work. It complains about a checksum mismatch.
I tried changing it to the one shown in the Detekt releases page but it didn't help. The checksum from the GraphQL query seems to be null.

## Context
I started hacking around but it became unteneable. We have numerous repos that would all need this patch which makes it too complicated.

## Your Environment
Github security scanning action.

**Comment by cortinico** [comment]
> The gh graphql script to download the release artifacts doesn't seem to work

Which script are you referring to?

**Comment by hjizettle** [comment]
In the setup procedure on the github security scanning there is a yml file pregenerated. It has inline graphql code that extracts the checksums and the release artifacts.

**Comment by cortinico** [comment]
> In the setup procedure on the github security scanning there is a yml file pregenerated. It has inline graphql code that extracts the checksums and the release artifacts.

Sorry could you link which instructions have you followed?
Can you also share the pregenerated yaml file?

What is actionable on our end?

**Comment by hjizettle** [comment]
I am sorry I really don't know who manages these yml files. In our repos Security tab there is a code scanning section and when I choose to configure `detekt` the page presents me with a yml file (more or less a github action config I think).
You (or someone who is an admin) should be able to see it under https://github.com/detekt/detekt/security for the detekt project itself. I think it's available for all github projects just a mater of configuring it.

**Comment by 3flex** [comment]
Please post a link to the instructions you're following - as mentioned above we'll need more information to understand exactly what you're trying to do, as it's possible that none of the maintainers are familiar with the feature you're trying to set up, so may not be able to help without more information.

Screenshots may also help.

**Comment by hjizettle** [comment]
There is a "detekt" box available in the regular Github security config for all repos I think. As a part of the setup a config file is generated which simply doesn't work and there are no obvious instructions on what is needed. There is really no more information to provide since I got no further. I thought someone on the Detekt team/project had created this config in the first place but perhaps that is not so? 

There are related issues/docs/blogs from before so Github code scanning is used:
https://github.com/detekt/detekt/issues/3045
https://github.com/marketplace/actions/detekt-all

The generated yml that doesn't work looks like this. It's a common Github action I believe but having to edit it extensively kind of defeats the purpose of having it as basically a "one click to go" button in the Github UI.

```yml
# This workflow uses actions that are not certified by GitHub.
# They are provided by a third-party and are governed by
# separate terms of service, privacy policy, and support
# documentation.

# This workflow performs a static analysis of your Kotlin source code using
# Detekt.
#
# Scans are triggered:
# 1. On every push to default and protected branches
# 2. On every Pull Request targeting the default branch
# 3. On a weekly schedule
# 4. Manually, on demand, via the "workflow_dispatch" event
#
# The workflow should work with no modifications, but you might like to use a
# later version of the Detekt CLI by modifing the $DETEKT_RELEASE_TAG
# environment variable.
name: Scan with Detekt

on:
  # Triggers the workflow on push or pull request events but only for default and protected branches
  push:
    branches: [ "main" ]
  pull_request:
    branches: [ "main" ]
  schedule:
     - cron: '18 17 * * 3'

  # Allows you to run this workflow manually from the Actions tab
  workflow_dispatch:

env:
  # Release tag associated with version of Detekt to be installed
  # SARIF support (required for this workflow) was introduced in Detekt v1.15.0
  DETEKT_RELEASE_TAG: v1.21.0
  EXPECTED_DETEKT_RELEASE_SHA: 32f6e22d9524804ebbb51d31e53cd28a84864ed6

# A workflow run is made up of one or more jobs that can run sequentially or in parallel
jobs:
  # This workflow contains a single job called "scan"
  scan:
    name: Scan
    # The type of runner that the job will run on
    runs-on: ubuntu-latest

    # Steps represent a sequence of tasks that will be executed as part of the job
    steps:
    # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
    - uses: actions/checkout@v3

    # Gets the download URL associated with the $DETEKT_RELEASE_TAG
    - name: Get Detekt download URL
      id: detekt_info
      env:
        GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
      run: |
        gh api graphql --field tagName=$DETEKT_RELEASE_TAG --raw-field query='
          query getReleaseAssetDownloadUrl($tagName: String!) {
            repository(name: "detekt", owner: "detekt") {
              release(tagName: $tagName) {
                releaseAssets(name: "detekt", first: 1) {
                  nodes {
                    downloadUrl
                  }
                }
                tagCommit {
                  oid
                }
              }
            }
          }
        ' 1> gh_response.json
        
        DETEKT_RELEASE_SHA=$(jq --raw-output '.data.repository.release.releaseAssets.tagCommit.oid' gh_response.json)
        if [ $DETEKT_RELEASE_SHA != $EXPECTED_DETEKT_RELEASE_SHA ]; then
          echo "Release tag doesn't match expected $EXPECTED_DETEKT_RELEASE_SHA commit SHA but got $DETEKT_RELEASE_SHA"
          exit 1
        fi
        
        DETEKT_DOWNLOAD_URL=$(jq --raw-output '.data.repository.release.releaseAssets.nodes[0].downloadUrl' gh_response.json)
        echo "::set-output name=download_url::$DETEKT_DOWNLOAD_URL"

    # Sets up the detekt cli
    - name: Setup Detekt
      run: |
        dest=$( mktemp -d )
        curl --request GET \
          --url ${{ steps.detekt_info.outputs.download_url }} \
          --silent \
          --location \
          --output $dest/detekt
        chmod a+x $dest/detekt
        echo $dest >> $GITHUB_PATH

    # Performs static analysis using Detekt
    - name: Run Detekt
      continue-on-error: true
      run: |
        detekt --input ${{ github.workspace }} --report sarif:${{ github.workspace }}/detekt.sarif.json

    # Modifies the SARIF output produced by Detekt so that absolute URIs are relative
    # This is so we can easily map results onto their source files
    # This can be removed once relative URI support lands in Detekt: https://git.io/JLBbA
    - name: Make artifact location URIs relative
      continue-on-error: true
      run: |
        echo "$(
          jq \
            --arg github_workspace ${{ github.workspace }} \
            '. | ( .runs[].results[].locations[].physicalLocation.artifactLocation.uri |= if test($github_workspace) then .[($github_workspace | length | . + 1):] else . end )' \
            ${{ github.workspace }}/detekt.sarif.json
        )" > ${{ github.workspace }}/detekt.sarif.json

    # Uploads results to GitHub repository using the upload-sarif action
    - uses: github/codeql-action/upload-sarif@v2
      with:
        # Path to SARIF file relative to the root of the repository
        sarif_file: ${{ github.workspace }}/detekt.sarif.json
        checkout_path: ${{ github.workspace }}
```

**Comment by 3flex** [comment]
Ok I understand now.

To my knowledge that config wasn't created by the detekt maintainers. I also did a Google search and a GitHub search for `EXPECTED_DETEKT_RELEASE_SHA` but there are no matches on either. I was hoping the original source might be revealed and we could point you to the right issue tracker.

I'm not sure how to help with this, hopefully someone else knows what's responsible for generating the config and where to send issues to have that fixed.

**Comment by 3flex** [comment]
I'm closing this for now as it's not actionable by this project as far as I can tell... If there's more information or someone can direct us to the source of this config we might be able to work with them on updating it and getting a working config.

**Comment by JordanLongstaff** [comment]
Is it possible to get the latest release tag for Detekt, e.g. via a cURL request?

**Comment by 3flex** [comment]
I assume there's a way to get it from the GitHub API.
